MimeKit/Cryptography/BouncyCastleSecureMimeContext.cs

```diff
@@ -694,20 +694,41 @@ X509Certificate GetCertificate (IStore<X509Certificate> store, SignerID signer)
 		/// <returns>The certificate chain, including the specified certificate.</returns>
 		protected IList<X509Certificate> BuildCertificateChain (X509Certificate certificate)
 		{
-			var selector = new X509CertStoreSelector {
-				Certificate = certificate
-			};
+			var selector = new X509CertStoreSelector ();
 
-			var intermediates = new X509CertificateStore ();
-			intermediates.Add (certificate);
+			var userCertificateStore = new X509CertificateStore ();
+			userCertificateStore.Add (certificate);
 
-			var parameters = new PkixBuilderParameters (GetTrustedAnchors (), selector) {
+			var issurerStore = GetTrustedAnchors ();
+			var anchorStore = new X509CertificateStore ();
+			
+			foreach (var anchor in issurerStore) {
+				anchorStore.Add (anchor.TrustedCert);
+			}
+
+			var parameters = new PkixBuilderParameters (issurerStore, selector) {
 				ValidityModel = PkixParameters.PkixValidityModel,
-				IsRevocationEnabled = false,
+				IsRevocationEnabled = CheckCertificateRevocation,
 				Date = DateTime.UtcNow
```

Comment:
I believe we should validate during the signing process. This option is essential for my workflows. One of the key advantages of SMIME is the ability to revoke a bad actor at scale. While this might not be as evident in scenarios where certificates are exchanged manually, it makes a lot of sense in my environment where the public certificate is discoverable via DNS.

I fully expected you to push back on the missing async paths.


I just got to a place where you could see the theme of the PR and tests to support it.  I am used to writing the chain validation with ```System.Security.Cryptography```.  And there is just a lot going on in here to get familiar with.  The thing I like about the BouncyCastle technique is I can control the caching. where with the MS stack on Windows it puts discovered intermediates in the machine store and crls in the file system and caches in memory.  You can control the machine store and file system but you have to restart the service to clear the in memory cache.  Maybe there is a PInvoke method that could punt the cache but I am not sure.